java/algorithm/src/main/java/org/apache/arrow/algorithm/hash/ArrowBufHasher.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.algorithm.hash;
+
+import java.nio.ByteOrder;
+
+import org.apache.arrow.util.Preconditions;
+
+import io.netty.buffer.ArrowBuf;
+
+/**
+ * Utility for calculating the hash code for a consecutive memory region.
+ */
+public abstract class ArrowBufHasher {
+
+  public static final boolean LITTLE_ENDIAN = ByteOrder.nativeOrder() == ByteOrder.LITTLE_ENDIAN;
+
+  /**
+   * Calculates the hash code for a memory region.
+   * @param buf the buffer for the memory region.
+   * @param offset offset within the buffer for the memory region.
+   * @param length length of the memory region.
+   * @return the hash code.
+   */
+  public int hashCode(ArrowBuf buf, int offset, int length) {
+    Preconditions.checkArgument(offset >= 0 && length >= 0 && offset + length <= buf.capacity());
+
+    int hashValue = 0;
+    int index = 0;
+    while (index + 8 <= length) {
+      long longValue = buf.getLong(index + offset);
```

Comment:
can be replaced with PlatformDependent.getLong() interprets according to the platform endianess

Reply:
@praveenbingo thanks a lot for your comments.

Here we force the algorithm to interpret the data in little endian, in a platform independent way. This is to make sure that, if the data is sent from a little endian machine to a big endian machine, its hash code remain unchanged.

Reply:
I see, sounds good.